cli/tools/fmt.rs

```diff
@@ -501,7 +509,50 @@ pub fn format_html(
   })
 }
 
-/// Formats a single TS, TSX, JS, JSX, JSONC, JSON, MD, or IPYNB file.
+pub fn format_sql(
+  file_text: &str,
+  fmt_options: &FmtOptionsConfig,
+) -> Result<Option<String>, AnyError> {
+  let ignore_file = file_text
+    .lines()
+    .take_while(|line| line.starts_with("--"))
+    .any(|line| {
+      line
+        .strip_prefix("--")
+        .unwrap()
+        .trim()
+        .starts_with("deno-fmt-ignore-file")
+    });
+
+  if ignore_file {
+    return Ok(None);
+  }
+
+  let mut formatted_str = sqlformat::format(
+    file_text,
+    &sqlformat::QueryParams::None,
+    &sqlformat::FormatOptions {
+      ignore_case_convert: None,
+      indent: if fmt_options.use_tabs.unwrap_or_default() {
+        sqlformat::Indent::Tabs
+      } else {
+        sqlformat::Indent::Spaces(fmt_options.indent_width.unwrap_or(2))
+      },
+      // leave one blank line between queries.
+      lines_between_queries: 2,
+      uppercase: Some(true),
+    },
+  )
+  .trim_end()
+  .to_owned();
+
+  // Add single new line to the end of file.
+  formatted_str.push_str("\n");
+
+  Ok(Some(formatted_str.to_owned()))
```

Comment:
It seems we're allocating twice here. Do we really need to `trim_end()` and then push a new line?

Reply:
It appers that sqlformat does not add a new line at the end and there is no option to control that.

In my opinion, it would preferably be better to have this supported by sqlformat directly, i did it here because it does not support.

Gonna open an issue with sqlformat to see if this is something they would like to have support for.

Reply:
I guess it's fine then. Maybe just remove the first `to_owned()` since `sqlformat` already returns an owned `String`.

Reply:
Already did.